### PR TITLE
ignore functions with more than one element ...

### LINKED ISSUE #3583: L030: built-in tsql functions should be capitalized by L030 while protecting user defined functions
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

Looking at my code formatted with sqlfluff so far, 

I found that for example `month()`, `day()`, `year()`, `sum()`, `iif()` are not recognized as keywords and thus not capitalized yet.

As a result, I have gone through https://docs.microsoft.com/en-us/sql/t-sql/functions/functions?view=sql-server-ver16  and created an initial list of t-sql functions:

```
ASCII
CHAR
CHARINDEX
CONCAT
CONCAT_WS
DIFFERENCE
FORMAT
LEFT
LEN
LOWER
LTRIM
NCHAR
PATINDEX
QUOTENAME
REPLACE
REPLICATE
REVERSE
RIGHT
RTRIM
SOUNDEX
SPACE
STR
STRING_AGG
STRING_ESCAPE
STRING_SPLIT
STUFF
SUBSTRING
TRANSLATE
TRIM
UNICODE
UPPER

CAST 
CONVERT
PARSE
TRY_CAST
TRY_CONVERT
TRY_PARSE
CURSOR_STATUS

SYSDATETIME 
SYSDATETIMEOFFSET
SYSUTCDATETIME
GETDATE
GETUTCDATE
DATE_BUCKET
DATENAME 
DATEPART 
DAY 
MONTH 
YEAR 
	
DATEFROMPARTS 
DATETIME2FROMPARTS 
DATETIMEFROMPARTS 
DATETIMEOFFSETFROMPARTS 
SMALLDATETIMEFROMPARTS 
TIMEFROMPARTS 
DATEDIFF 
DATEDIFF_BIG 
DATEADD 
EOMONTH 
SWITCHOFFSET 
TODATETIMEOFFSET 
ISDATE 
	
ISJSON
JSON_VALUE
JSON_QUERY
JSON_MODIFY
JSON_PATH_EXISTS

CHOOSE 
IIF 

ABS
ACOS
ASIN
ATAN
ATN2
CEILING
COS
COT
DEGREES
EXP
FLOOR
LOG
LOG10
PI
POWER
RADIANS  
RAND  
ROUND  
SIGN  
SIN  
SQRT  
SQUARE  
TAN

APP_NAME
APPLOCK_MODE
APPLOCK_TEST
ASSEMBLYPROPERTY
COL_LENGTH
COL_NAME
COLUMNPROPERTY
DATABASE_PRINCIPAL_ID
DATABASEPROPERTYEX
DB_ID
DB_NAME
FILE_ID
FILE_IDEX
FILE_NAME
FILEGROUP_ID
FILEGROUP_NAME
FILEGROUPPROPERTY
FILEPROPERTY
FULLTEXTCATALOGPROPERTY
FULLTEXTSERVICEPROPERTY
INDEX_COL  
INDEXKEY_PROPERTY  
INDEXPROPERTY  
NEXT VALUE FOR  
OBJECT_DEFINITION  
OBJECT_ID  
OBJECT_NAME  
OBJECT_SCHEMA_NAME  
OBJECTPROPERTY  
OBJECTPROPERTYEX  
ORIGINAL_DB_NAME  
PARSENAME  
SCHEMA_ID  
SCHEMA_NAME  
SCOPE_IDENTITY  
SERVERPROPERTY  
STATS_DATE  
TYPE_ID  
TYPE_NAME  
TYPEPROPERTY  
VERSION  

CERTENCODED
PWDCOMPARE
CERTPRIVATEKEY
PWDENCRYPT
CURRENT_USER
SCHEMA_ID
DATABASE_PRINCIPAL_ID
SCHEMA_NAME
SESSION_USER
SUSER_ID
SUSER_SID
HAS_PERMS_BY_NAME
SUSER_SNAME
IS_MEMBER
SYSTEM_USER
IS_ROLEMEMBER
SUSER_NAME
IS_SRVROLEMEMBER
USER_ID
LOGINPROPERTY
USER_NAME
ORIGINAL_LOGIN
PERMISSIONS

TEXTPTR

CUME_DIST
CURRENT_TIMESTAMP
DENSE_RANK
FIRST_VALUE
FORMAT
GETDATE
GETUTCDATE
GET_TRANSMISSION_STATUS
LAG
LAST_VALUE
LEAD
MIN_ACTIVE_ROWVERSION
NEWID
NEWSEQUENTIALID
NEXT VALUE FOR
NTILE
PARSENAME
PERCENTILE_CONT
PERCENTILE_DISC
PERCENT_RANK
RAND
RANK
ROW_NUMBER
SUM
```

Once an initial set of the above is integrated, i can certainly validate it once more and create a PR for all missing items.

### Use case

easier to read code.

### Dialect

tsql

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by fmms**
### Brief summary of the change made
fixes #3583 

### Pull Request checklist
- [ ] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/rules/L010.py [hunk 1]
```diff
@@ -78,6 +78,15 @@ def _eval(self, context: RuleContext) -> Optional[List[LintResult]]:
         if self.matches_target_tuples(context.segment, self._exclude_elements, parent):
             return [LintResult(memory=context.memory)]
 
+        # If the parent element is a function_name and there is more than one element
+        # below it, we ignore it because it is most likely a user defined function.
+        # SUM() should be modified. However, dbo.myScalar() must not be modified.
```
Comment (tunetheweb) [inline src/sqlfluff/rules/L010.py:83]: I found this difficult to follow. Would it be easier to just say something like this:

```suggestion
        # Used by L030 (that inherits from this rule)
        # If it's a qualified function_name (i.e with more than one part to
        # function_name). Then it is likely an existing user defined function (UDF)
        # which are case sensitive so ignore for this.
```

I wonder should there be a `include_qualified_functions` option in this (defaulted to `false`)?

Also, I think in some SQL dialects there might be a `SYS` schema or such like that contains the system functions, which are case insensitive? Then again not sure if there's a good reason to use that, so maybe fine to ignore that edge case.

File: test/fixtures/rules/std_rule_cases/L030.yml [hunk 2]
```diff
@@ -42,8 +42,7 @@ test_fail_capitalization_after_comma:
   fix_str: SELECT FLOOR(dt) ,COUNT(*) FROM test
 
 test_fail_fully_qualified_function_mixed_functions:
```
Comment (tunetheweb) [inline test/fixtures/rules/std_rule_cases/L030.yml:44]:
```suggestion
test_pass_fully_qualified_function_mixed_functions:
```

File: test/fixtures/rules/std_rule_cases/L030.yml [hunk 3]
```diff
@@ -63,8 +62,7 @@ test_pass_ignore_templated_code:
 
 test_fail_func_name_templated_literal_mix:
```
Comment (tunetheweb) [inline test/fixtures/rules/std_rule_cases/L030.yml:63]:
```suggestion
test_pass_func_name_templated_literal_mix:
```

**Review by tunetheweb — CHANGES_REQUESTED** [review]

**Comment by barrywhart** [comment]
@tunetheweb: Could you take another look at this since you'd requested changes on it?

**Review by tunetheweb — APPROVED** [review]
